src/zcbenchmarks.cpp

```diff
@@ -42,64 +44,52 @@ double benchmark_sleep()
 double benchmark_parameter_loading()
 {
     // FIXME: this is duplicated with the actual loading code
-    boost::filesystem::path pk_path = ZC_GetParamsDir() / "zc-testnet-public-alpha-proving.key";
-    boost::filesystem::path vk_path = ZC_GetParamsDir() / "zc-testnet-public-alpha-verification.key";
+    boost::filesystem::path pk_path = ZC_GetParamsDir() / "z3-proving.key";
+    boost::filesystem::path vk_path = ZC_GetParamsDir() / "z3-verification.key";
 
     timer_start();
-    auto vk_loaded = libzerocash::ZerocashParams::LoadVerificationKeyFromFile(
-        vk_path.string(),
-        INCREMENTAL_MERKLE_TREE_DEPTH
-    );
-    auto pk_loaded = libzerocash::ZerocashParams::LoadProvingKeyFromFile(
-        pk_path.string(),
-        INCREMENTAL_MERKLE_TREE_DEPTH
-    );
-    libzerocash::ZerocashParams zerocashParams = libzerocash::ZerocashParams(
-        INCREMENTAL_MERKLE_TREE_DEPTH,
-        &pk_loaded,
-        &vk_loaded
-    );
-    return timer_stop();
-}
 
-double benchmark_create_joinsplit()
-{
-    CScript scriptPubKey;
+    auto newParams = ZCJoinSplit::Unopened();
 
-    std::vector<PourInput> vpourin;
-    std::vector<PourOutput> vpourout;
+    newParams->loadVerifyingKey(vk_path.string());
+    newParams->preloadProvingKey(pk_path.string());
+    newParams->loadProvingKey();
 
-    while (vpourin.size() < NUM_POUR_INPUTS) {
-        vpourin.push_back(PourInput(INCREMENTAL_MERKLE_TREE_DEPTH));
-    }
+    double ret = timer_stop();
 
-    while (vpourout.size() < NUM_POUR_OUTPUTS) {
-        vpourout.push_back(PourOutput(0));
-    }
+    delete newParams;
+
+    return ret;
+}
+
+double benchmark_create_joinsplit()
+{
+    // TODO: #808
+    uint256 pubKeyHash;
 
     /* Get the anchor of an empty commitment tree. */
-    IncrementalMerkleTree blank_tree(INCREMENTAL_MERKLE_TREE_DEPTH);
-    std::vector<unsigned char> newrt_v(32);
-    blank_tree.getRootValue(newrt_v);
-    uint256 anchor = uint256(newrt_v);
+    uint256 anchor = ZCIncrementalMerkleTree().root();
 
     timer_start();
-    CPourTx pourtx(*pzerocashParams,
-                   scriptPubKey,
+    CPourTx pourtx(*pzcashParams,
+                   pubKeyHash,
                    anchor,
-                   {vpourin[0], vpourin[1]},
-                   {vpourout[0], vpourout[1]},
+                   {JSInput(), JSInput()},
+                   {JSOutput(), JSOutput()},
                    0,
                    0);
     double ret = timer_stop();
-    assert(pourtx.Verify(*pzerocashParams));
+
+    assert(pourtx.Verify(*pzcashParams, pubKeyHash));
     return ret;
 }
 
 double benchmark_verify_joinsplit(const CPourTx &joinsplit)
 {
     timer_start();
-    joinsplit.Verify(*pzerocashParams);
+    // TODO: #808
+    uint256 pubKeyHash;
```

Comment:
This times the `uint256` constructor, which I think is fast, but may be non-negligible compared to `joinsplit.Verify()`. Move this line above the `timer_start()`.